src/createServer.js

```diff
@@ -1,11 +1,184 @@
 'use strict';
 
-// const express = require('express');
+const express = require('express');
+
+const { Expenses } = require('./services/Expenses.js');
+const { Users } = require('./services/Users.js');
 
 function createServer() {
-  // Use express to create a server
-  // Add a routes to the server
-  // Return the server (express app)
+  const app = express();
+
+  const expenses = new Expenses();
+  const users = new Users();
+
+  app.get('/users', (req, res) => {
+    res.send(users.getAll());
+  });
+
+  app.get('/users/:userId', (req, res) => {
+    const { userId } = req.params;
+
+    const user = users.getOne(userId);
+
+    if (!user) {
+      res.sendStatus(404);
+
+      return;
+    }
+
+    res.send(user);
+  });
+
+  app.post('/users', express.json(), (req, res) => {
+    const { name } = req.body;
+
+    if (!name) {
+      res.sendStatus(400);
+
+      return;
+    }
+
+    res.statusCode = 201;
+    res.send(users.create(name));
+  });
+
+  app.delete('/users/:userId', (req, res) => {
+    const { userId } = req.params;
+
+    const userToDelete = users.getOne(userId);
+
+    if (!userToDelete) {
+      res.sendStatus(404);
+
+      return;
+    }
+
+    users.deleteOne(userId);
+    res.sendStatus(204);
+  });
+
+  app.patch('/users/:userId', express.json(), (req, res) => {
```

Comment:
add app.use(express.json()), and avoid duplicates :)

Reply:
done!